modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ *
+ * @return array $metadata
+ * @since n.e.x.t
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
```

Comment:
Please avoid blank lines between hook callback functions and their `add_action` / `add_filter` call.
```suggestion
}
add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
```